.github/workflows/build.yml

```diff
@@ -137,14 +155,14 @@ jobs:
         run: |
           make cache-export-docker-load-all
       - name: clear linuxkit cache so we can load for target arch
-        if: ${{ matrix.arch != 'amd64' }}  # because our runner arch is amd64; if that changes, this will have to change
+        if: ${{ steps.arch_runner_equals_matrix.outputs.matched != 'true' }}
```

Comment:
hold on, we're always running this job on ubuntu-22.04, which means amd64, so we will cross-compile for arm, I thought that this check should be just avoided and we can just pull the cache for respective architecture we're building, because if we want to run this on the same architectures, it might make more sense to combine those two...

Reply:
Agreed, better safe than sorry in this case